### PR TITLE
Dedup src diffs between 2.10 & 2.11+

### LINKED ISSUE #233: Minimise the source code differences between 2.10 and non-2.10 compiler bridge

### PR DESCRIPTION
**Pull request by dwijnand**
Fixes #233

### CODE REVIEW

**Comment by dwijnand** [comment]
I dropped https://github.com/sbt/zinc/pull/244/commits/a947ac8cd9bb891f8df05a23eae8c000fed6f054 which makes reviewing 2.10 Compat changes much easier.

File: internal/compiler-bridge/src/main/scala_2.10/xsbt/Compat.scala
Comment (jvican) [inline internal/compiler-bridge/src/main/scala_2.10/xsbt/Compat.scala]: Can we make this private?

File: internal/compiler-bridge/src/main/scala_2.10/xsbt/Compat.scala
Comment (jvican) [inline internal/compiler-bridge/src/main/scala_2.10/xsbt/Compat.scala]: `@inline` here perhaps? Scalac is very pessimistic about inlining, but it's better to have it than not to have it.

File: internal/compiler-bridge/src/main/scala_2.10/xsbt/Compat.scala
Comment (jvican) [inline internal/compiler-bridge/src/main/scala_2.10/xsbt/Compat.scala]: Can we annotate these two also with `@inline`? Just in case, same comment as `unexpandedName`.

File: internal/compiler-bridge/src/main/scala_2.10/xsbt/Compat.scala
Comment (jvican) [inline internal/compiler-bridge/src/main/scala_2.10/xsbt/Compat.scala:134]: Good work here.

**Review by jvican — APPROVED** [review]
This is great stuff, thanks for it @dwijnand.
  
Changes are much less intrusive of what I thought, and if they happen they will only affect 2.10 sources, so this patch has my blessing. I don't think we need benchmarks to get this in.
  
I'll rebase my other PR on top of this one, I really don't want to migrate all my stuff to 2.10.

File: internal/compiler-bridge/src/main/scala_2.10/xsbt/Compat.scala
Comment (dwijnand) [reply internal/compiler-bridge/src/main/scala_2.10/xsbt/Compat.scala]: Sure I can. I made it public because it's public in the original (2.11+) API.

**Review by dwijnand — COMMENTED** [review]
I'll make the forwarders final and inline.

**Comment by dwijnand** [comment]
This is green now, after a few brain farts and a moment of IO flakiness.

**Comment by eed3si9n** [comment]
Nice work. Could you rebase against the master plz?

File: internal/compiler-bridge/src/main/scala_2.10/xsbt/Compat.scala
Comment (romanowski) [inline internal/compiler-bridge/src/main/scala_2.10/xsbt/Compat.scala:127]: extends AnyVal?

**Review by romanowski — APPROVED** [review]
Good work!

File: internal/compiler-bridge/src/main/scala_2.10/xsbt/Compat.scala
Comment (dwijnand) [reply internal/compiler-bridge/src/main/scala_2.10/xsbt/Compat.scala:127]: Value classes can't be nested, and MacroExpansion depends on global.

Comment (romanowski) [reply internal/compiler-bridge/src/main/scala_2.10/xsbt/Compat.scala:127]: ah, ok

**Comment by dwijnand** [comment]
@jvican ExtractUsedNamesSpecification is failing. Do you mind having a look?

**Comment by jvican** [comment]
Not at all! @dwijnand Will keep you posted tomorrow, I won't be very available but I'll try to have a look at this.

**Comment by jvican** [comment]
This is weird. I cannot reproduce locally from your branch.

**Comment by jvican** [comment]
No, wait, just reproduced.

**Comment by jvican** [comment]
Just opened https://github.com/dwijnand/sbt-zinc/pull/2 to merge my fixes into dwijand to pass the CI here.

**Comment by jvican** [comment]
CI passing. Waiting for another Zinc maintainer to sign off.

**Review by eed3si9n — APPROVED** [review]

**Pull request merged**
